File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> st = new Stack<Integer>();
        for (int i = 0;i<tokens.length;i++)
        {
            if (tokens[i].equals("+"))
            {
                int b = st.peek();
                st.pop();
                int a  = st.peek();
                st.pop();
                int c = a + b;
                st.push(c);
            }
            else if (tokens[i].equals("-"))
            {
                int b = st.peek();
                st.pop();
                int a  = st.peek();
                st.pop();
                int c = a - b;
                st.push(c);
            }
            else if (tokens[i].equals("*"))
            {
                int b = st.peek();
                st.pop();
                int a  = st.peek();
                st.pop();
                int c = a * b;
                st.push(c);
            }
            else if (tokens[i].equals("/"))
            {
                int b = st.peek();
                st.pop();
                int a  = st.peek();
                st.pop();
                int c = a / b;
                st.push(c);
            }
            else {
                st.push(Integer.parseInt(tokens[i]));
            }
        }
        return st.peek();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for token in tokens:
            stack.append(token)
        def callstack() -> int:
            nonlocal stack
            operation = stack.pop()
            if operation.isdigit():
                return int(operation)
            if len(operation) > 1 and operation[0] == "-":
                return int(operation)
            vala = callstack()
            valb = callstack()
            if operation == "+":
                return valb + vala
            elif operation == "-":
                return valb - vala
            elif operation == "*":
                return valb * vala
            else:
                return int(valb / vala)
        return callstack()